Description: fix hard-coded database locations
Author: Lars Kruse <devel@sumpfralle.de>
Applied-Upstream: https://github.com/munin-monitoring/munin/commit/c8148a992637e365b7fa6329c7cb4f2d30c0735f
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/lib/Munin/Master/HTML.pm
+++ b/lib/Munin/Master/HTML.pm
@@ -839,7 +839,7 @@ sub get_param
 	# Ok, now SQL is needed to go further
         use DBI;
 	my $datafilename = $ENV{MUNIN_DBURL} || "$Munin::Common::Defaults::MUNIN_DBDIR/datafile.sqlite";
-        my $dbh = DBI->connect("dbi:Pg:dbname=munin","","") or die $DBI::errstr;
+	my $dbh = Munin::Master::Update::get_dbh();
 
 	my ($value) = $dbh->selectrow_array("SELECT value FROM param WHERE name = ?", undef, ($param));
 
--- a/lib/Munin/Master/LimitsOld.pm
+++ b/lib/Munin/Master/LimitsOld.pm
@@ -326,8 +326,8 @@ sub process_service {
     my $datafilename = $ENV{MUNIN_DBURL} || $config->{dbdir}."/datafile.sqlite";
     my $datafilename_state = $ENV{MUNIN_DBURL} || $config->{dbdir}."/datafile-state.sqlite";
     DEBUG "[DEBUG] opening sql $datafilename";
-    my $dbh = DBI->connect("dbi:SQLite:dbname=$datafilename","","") or die $DBI::errstr;
-    my $dbh_state = DBI->connect("dbi:SQLite:dbname=$datafilename","","") or die $DBI::errstr;
+    my $dbh = Munin::Master::Update::get_dbh() or die $DBI::errstr;
+    my $dbh_state = Munin::Master::Update::get_dbh() or die $DBI::errstr;
     my $sth_state = $dbh_state->prepare('SELECT last_epoch, last_value, prev_epoch, prev_value, alarm FROM state WHERE id = ? and type = ?');
     my $sth_state_upt = $dbh_state->prepare('UPDATE state SET alarm = ? WHERE id = ? and type = ?');
     my $sth_ds = $dbh->prepare('SELECT id FROM url WHERE path = ? and type = ?');
